src/views/ArticleView.vue

```diff
@@ -0,0 +1,303 @@
+				</ul>
+			</ul>
+		</div>
+	</div>
+</template>
+
+<script setup>
+import { useRoute } from 'vue-router';
+import data from '../data.json';
+const route = useRoute();
+
+const transforms = {
+	'put styles in body': ( doc ) => {
+		for ( const style of doc.head.querySelectorAll( 'link[rel="stylesheet"]' ) ) {
+			doc.body.prepend( style );
+		}
+	},
+	'remove stuff': ( doc ) => {
+		const selector = [
+			'script',
+			'sup',
+			'.pcs-collapse-table-container',
+			'.hatnote',
+			"[ role='navigation' ]"
+		].join( ',' );
+		for ( const n of doc.querySelectorAll( selector ) ) {
+			n.remove();
+		}
+	},
+	'turn links into plain text': ( doc ) => {
+		for ( const a of doc.querySelectorAll( 'a' ) ) {
+			a.replaceWith( a.innerHTML );
+		}
+	},
+	'remove sections after fold': ( doc ) => {
+		for ( const section of doc.querySelectorAll( '.pcs-fold-hr ~ section' ) ) {
+			section.remove();
+		}
+	},
+	'remove "See also" section': ( doc ) => {
+		const seeAlsoH2 = doc.querySelector( '#See_also' );
+		if ( seeAlsoH2 ) {
+			seeAlsoH2.parentElement.parentElement.remove();
+		}
+	},
+	'turn figure into an image': ( doc ) => {
+		for ( const figure of doc.querySelectorAll( 'figure' ) ) {
+			const newSpanElement = document.createElement( 'span' );
+			const imgElement = document.createElement( 'img' );
+			imgElement.className = 'figure';
+			newSpanElement.innerHTML = figure.innerText;
+
+			// Set the source attribute of the <img>
+			if ( newSpanElement.firstChild && newSpanElement.firstChild.dataset ) {
+				imgElement.src = newSpanElement.firstChild.dataset.src ||
+					newSpanElement.firstChild.src;
+				figure.firstChild.remove();
+				figure.insertBefore( imgElement, figure.firstChild );
+			}
+		}
+	},
+	'turn sections into collapsible section': ( doc ) => {
+		const sections = doc.querySelectorAll( 'section[data-mw-section-id]' );
+
+		for ( let i = 1; i < sections.length; i++ ) {
+			const section = sections[ i ];
+			if ( section.parentElement.className.indexOf( 'collapsible' ) === -1 ) {
+				section.style.display = 'unset';
+				section.classList.add( 'collapsible' );
+			} else {
+				section.style.display = null;
+			}
+		}
+	}
+};
+
+const addEvents = function ( doc ) {
+	// add sections collapsible toggle events
+	for ( const section of doc.querySelectorAll( '.collapsible' ) ) {
+		section.firstChild.onclick = function () {
+			section.classList.toggle( 'collapsible' );
+		};
+	}
+};
+
+const fetchArticle = function ( title ) {
+	const url = `https://en.wikipedia.org/api/rest_v1/page/mobile-html/${title}`;
+
+	return fetch( url ).then( ( resp ) => {
+		if ( resp.ok ) {
+			return resp.text();
+		} else {
+			throw new Error();
+		}
+	} ).then( ( html ) => {
+
+		// transform response html object
+		const doc = new DOMParser().parseFromString( html, 'text/html' );
+		Object.keys( transforms ).forEach( ( key ) => {
+			transforms[ key ]( doc );
+		} );
+
+		document.querySelector( '.content' ).innerHTML = doc.body.outerHTML;
```

Comment:
I think you should assign that html to local reactive variable so you can make everything appear all at once. You can see the rendering sequence by emulating a slow network... it's not pretty.